### PR TITLE
Add support for NIC resource

### PR DESCRIPTION
**Pull request by taohe1012**
##### SUMMARY

Add support for NIC resource

Relates OR Closes #0000

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

Resource: redfish_network_adapter

##### ADDITIONAL INFORMATION

At for nic resource are passed.
at-result
at-jobs

```paste below

```

#### Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

#### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

#### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: examples/resources/redfish_network_adapter/provider.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+terraform {
+  required_providers {
+    redfish = {
+      version = "1.3.0"
```
Comment (gokul-srivathsan) [inline examples/resources/redfish_network_adapter/provider.tf:21]: should be 1.4.0

Comment (taohe1012) [reply examples/resources/redfish_network_adapter/provider.tf]: thanks @gokul-srivathsan   updated

**Comment by taohe1012** [comment]
the resource_redfish_nic.go file is now divided to four files: `resource_redfish_nic.go` , `resource_redfish_nic_helper.go`, `resource_redfish_nic_schema.go`,` resource_redfish_nic_schema_helper.go`

AT is run again, and passed
image
image

**Review by rounak-adhikary — APPROVED** [review]

**Review by Krishnan-Priyanshu — APPROVED** [review]

**Pull request merged**